src/control/server/storage/provider.go

```diff
@@ -514,18 +514,22 @@ func (p *Provider) ScanBdevs(req BdevScanRequest) (*BdevScanResponse, error) {
 }
 
 // SetBdevCache stores given scan response in provider bdev cache.
-func (p *Provider) SetBdevCache(resp BdevScanResponse) error {
+func (p *Provider) SetBdevCache(resp *BdevScanResponse) error {
```

Comment:
I'm curious about the reasoning for this change, since it's turned into a copy at the end anyway.

Reply:
I need the changes in the response, which occur when filtering in this function, to be available to the caller of SetBdevCache (see usage in server_utils.go).

Reply:
yes this could be refactored to separate the responsibilities, looks like CI is not particularly stable at the moment so will have to re-run anyway

Reply:
done